File: programming problems/HackerRank/30 days of code/day 6, lets review/day_6_lets_review.py
from sys import stdin

input()
for line in stdin:

    if line == "":
        break

    word1, word2 = "", ""
    flag = True

    for ch in line.rstrip():
        if flag:
            word1 = "".join((word1, ch))
        else:
            word2 = "".join((word2, ch))
        flag = not flag

    print(f"{word1} {word2}")

File: programming problems/HackerRank/30 days of code/day 6, lets review/day_6_lets_review.js
function processData(input) {
    
    const words = input.split("\n")
    words.shift()

    for (let word of words) {
        
        let word1, word2 = word1 = "";
        let flag = true;

        [...word].forEach((ch, i) => {
            flag ? (word1 += ch) : (word2 += ch)
            flag = !flag
        })
        console.log(`${word1} ${word2}`);
    }
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
